src/CurrencyHelper.php

```diff
@@ -44,6 +44,32 @@ public function toFormatFromInt(int $value, string $currencyCode = CurrencyCode:
         return $this->toFormat($value / 100, $currencyCode, $locale);
     }
 
+    /**
+     * Transform an integer representing a decimal currency value (penny, cents...) into a monetary formatted string
+     * with the right currency symbol and the right localised format for the parameters respectively given.
+     *
+     * @param int $value
+     * @param string $currencyCode
+     * @param string $locale
+     *
+     * @return string
+     */
+    public function formatToMinorUnitWhenApplicable(int $value, string $currencyCode = CurrencyCode::POUND_STERLING, string $locale = 'en'): string
+    {
+        if ($value <= $this->getMinorUnitEnd($locale) && $pattern = $this->getMinorUnitPattern($locale)) {
```

Comment:
For clarity I recommend moving the `$pattern` declaration into its own line above

Reply:
a nested function?

Reply:
just `$pattern = $this->getMinorUnitPattern($locale);` in the line above the if statement and then just `&& $pattern` in the if statement

like that when you change the way you fetch the pattern it will be a different diff line than when you change anything else in the if statement

minor thing but helps with maintenance

Reply:
adding another line seems to be redundant, no?

if ($this->getMinorUnitPattern($locale)) {
$pattern = $this->getMinorUnitPattern($locale)
}

like this?

Reply:
```
$pattern = $this->getMinorUnitPattern($locale);
if ($value <= $this->getMinorUnitEnd($locale) && $pattern) {
    ...
```

Reply:
I don't think it changes much, but sure.